unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/SceneMetricsController.cs

```diff
@@ -270,46 +275,26 @@ protected void AddMetrics(DecentralandEntity entity)
                 entitiesMetrics[entity] = entityMetrics;
             }
 
-            if (VERBOSE) { Debug.Log("SceneMetrics: entity " + entity.entityId + " metrics " + entityMetrics.ToString()); }
+            if (VERBOSE)
+            {
+                Debug.Log("SceneMetrics: entity " + entity.entityId + " metrics " + entityMetrics.ToString());
+            }
 
             isDirty = true;
         }
 
         void CalculateMaterials(DecentralandEntity entity, EntityMetrics entityMetrics)
         {
-            //get boundaries checker if in debug cause it may have swap meshes materials
-            var debugBoundariesChecker = SceneController.i.boundariesChecker as SceneBoundariesDebugModeChecker;
-
-            //can we count materials directly from the mesh renderers?
-            bool isInValidPosition = debugBoundariesChecker == null
-                || (debugBoundariesChecker != null && debugBoundariesChecker.WasEntityInAValidPosition(entity));
+            var originalMaterials = Environment.i.sceneBoundsChecker.GetOriginalMaterials(entity.meshesInfo);
```

Comment:
I had to change the material fetch algorithm on this one, so this can be potentially slower and/or buggy.